bsp/Src/ADC.c

```diff
@@ -0,0 +1,163 @@
+#include "ADC.h"
+
+// Define Queue Handles
+#ifdef ADC1
+static ADC_HandleTypeDef hadc1_ = {.Instance = ADC1};
+ADC_HandleTypeDef* hadc1 = &hadc1_;
+#endif
+
+#ifdef ADC2
+static ADC_HandleTypeDef hadc2_ = {.Instance = ADC2};
+ADC_HandleTypeDef* hadc2 = &hadc2_;
+#endif
+
+#ifdef ADC3
+static ADC_HandleTypeDef hadc3_ = {.Instance = ADC3};
+ADC_HandleTypeDef* hadc3 = &hadc3_;
+#endif
+
+// Stores queue to push val to
+QueueHandle_t* adc1_q;
+QueueHandle_t* adc2_q;
+QueueHandle_t* adc3_q;
```

Comment:
These need to be Static Queues

Reply:
These are queue handles so I'm pretty sure they can be used for static queues as well. See [xQueueCreateStatic](https://freertos.org/Documentation/02-Kernel/04-API-references/06-Queues/02-xQueueCreateStatic),